### PR TITLE
Address feedback for PR: https://github.com/aspnet/Routing/pull/182

### PR DESCRIPTION
**Pull request by kichalla**
Related PR: https://github.com/aspnet/Routing/pull/182

@dougbu @Eilon

### CODE REVIEW

File: src/Microsoft.AspNet.Routing/RouteValueDictionary.cs [hunk 1]
```diff
@@ -31,23 +31,24 @@ public RouteValueDictionary()
         }
 
         /// <summary>
-        /// Creates a <see cref="RouteValueDictionary"/> initialized with the provided input value.
+        /// Creates a <see cref="RouteValueDictionary"/> initialized with the specified <paramref name="values"/>.
         /// </summary>
-        /// <param name="obj">An object to initialize the dictionary. The value can be of type
+        /// <param name="values">An object to initialize the dictionary. The value can be of type
         /// <see cref="IDictionary{TKey, TValue}"/> or <see cref="IReadOnlyDictionary{TKey, TValue}"/> or
         /// any other object.
```
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/RouteValueDictionary.cs]: still unsure what the second sentence adds.  my preference is to remove but I'd like to hear @Eilon's thoughts.

File: src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs [hunk 2]
```diff
@@ -414,7 +414,7 @@ private string DebuggerToString()
                 return string.Format(
                     "{{Accepted: '{0}' Filters: '{1}'}}",
                     string.Join(", ", _acceptedValues.Keys),
-                    string.Join(", ", _filters.Keys));
+                    string.Join(", ", _filters?.Keys));
```
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs]: :dancers:

File: test/Microsoft.AspNet.Routing.Tests/RouteValueDictionaryTests.cs [hunk 3]
```diff
@@ -168,36 +169,47 @@ public void CreateFromObject_MixedCaseThrows()
...
+
+                yield return new object[] { routeValues };
+            }
+        }
 
+        [Theory]
+        [MemberData(nameof(IEnumerableKeyValuePairData))]
+        public void RouteValueDictionary_CopiesValues_FromIEnumerableKeyValuePair(object values)
+        {
             // Act
```
Comment (dougbu) [inline test/Microsoft.AspNet.Routing.Tests/RouteValueDictionaryTests.cs]: Act -> Arrange & Act

File: test/Microsoft.AspNet.Routing.Tests/RouteValueDictionaryTests.cs [hunk 4]
```diff
@@ -168,36 +169,47 @@ public void CreateFromObject_MixedCaseThrows()
...
+        public void CreatedFrom_IEnumerableKeyValuePair_AllowsAddingOrModifyingValues(object values)
         {
-            // Arrange
-            var dictionary = new Dictionary<string, object>();
-            dictionary.Add("Name", "James");
-            dictionary.Add("Age", 30);
-            dictionary.Add("Address", new Address() { City = "Redmond", State = "WA" });
-            var readonlyDictionary = (IReadOnlyDictionary<string, object>)dictionary;
-
             // Act
```
Comment (dougbu) [inline test/Microsoft.AspNet.Routing.Tests/RouteValueDictionaryTests.cs]: Act -> Arrange & Act

**Comment by dougbu** [comment]
:watch: definitely looking good. mainly looking for a vote or two more about shortening the one `<param/>` description.

File: src/Microsoft.AspNet.Routing/RouteValueDictionary.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/RouteValueDictionary.cs]: from the original PR:
> We wanted to add it in the param description as remarks do not show up in intellisense.

Why do we want "The value can be of any type." to show up in IntelliSense?  And why waste the other words when the value can be of any type?

Comment (Eilon) [inline src/Microsoft.AspNet.Routing/RouteValueDictionary.cs]: I would change the second sentence to `or an object with public properties as key-value pairs.`

**Comment by kichalla** [comment]
Updated. /cc: @dougbu @Eilon

File: src/Microsoft.AspNet.Routing/RouteValueDictionary.cs [hunk 5]
```diff
@@ -31,23 +31,24 @@ public RouteValueDictionary()
...
+        /// <param name="values">An object to initialize the dictionary. The value can be of type
+        /// <see cref="IDictionary{TKey, TValue}"/> or <see cref="IReadOnlyDictionary{TKey, TValue}"/>
+        /// or an object with public properties as key-value pairs.
         /// </param>
         /// <remarks>
-        /// If the value is a dictionary, then its entries are copied. Otherwise the object is interpreted as a set
-        /// of key-value-pairs where the property names are keys, and property values are the values, and copied
-        /// into the dictionary. Only public instance non-index properties are considered.
+        /// If the value is a dictionary or other <see cref="IEnumerable{KeyValuePair{string, object}}"/>,
+        /// then its entries are copied. Otherwise the object is interpreted as a set of key-value pairs where the
```
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/RouteValueDictionary.cs]: suggest (not demand): everything else is now linked; so `object` -> `<see cref="object"/>` here and twice in the `<param/>` description above.

**Comment by dougbu** [comment]
:shipit:

File: src/Microsoft.AspNet.Routing/RouteValueDictionary.cs (hunk 5 above)
Comment (Eilon) [inline src/Microsoft.AspNet.Routing/RouteValueDictionary.cs]: I don't think linking to the `object` help doc helps much...

**Pull request merged**